### PR TITLE
Make layer order editable via drag and drop

### LINKED ISSUE #7100: Layer ordering for blend modes

### PR DESCRIPTION
**Pull request by MichaelBuessemeyer**
This PR makes the color layer settings sortable via drag. It adds a grabbing menu icon left to the enable/disable switch. Users can use this handle to adjust the order of the color layers which will affect the rendering output in case the cover mode is active. Afaik there is no difference for the additive blending mode.
The handle is only shown if there is more than one color layer.

### URL of deployed dev instance (used for testing):
- https://___.webknossos.xyz

##### Test 1:

- open a dataset with only one color layer (e.g. l4_sample in view mode or with a tracing)
- There should not be a handle next to the enable/disable switch of the color layer  

##### Test 2:

- Open an annotation of a dataset with multiple color layers
- Give different colors to the color layers
- Turn on cover mode
- Reposition the color layers. It should be possible to see the changes of the repositioning by observing the changes in color due to the color layer having different configured colors.
- Reload the annotation
- The order of the layers should have persisted and rendering should still work and look as expected.

### Issues:
- fixes #7100 

### TODOs:
- [x] Persist layer order settings
- [x] Consider making segmentation layer not movable in the first version as there is currently no use case :thinking: 

## Wanted Feedback:
- Currently, I called the field that saves the order of the color layers in the dataset configuration `layerOrder`. This is not as precise as it could be, because the variable only includes color layer names. Thus maybe `colorLayerOrder` would be more accurate and more understandable. But on the other hand, we might decide to also make the segmentation layer sortable and then the name would no longer be correct. I talked to Norman and his opinion is that the segmentation layer should not be sortable in the current state, as there is no real use-case right now for it and the segmentation layer will always be rendered on top; no matter whether one of the upper layers defines the pixel as covered by selecting cover blend mode or not. Therefore the current version does not include the segmentation layers being draggable / sortable. @philippotto what do you is the best name to keep for the field? In long term `layerOrder` might be smarter but more confusing right now and `colorLayerOrder` would be good at the moment, but bad once segmentation layers are also sortable which might come in the future.
- Update: I now renamed the field to `colorLayerOrder`

### Important Note
In case anyone comes across this pr while developing the pr that makes segmentation layers sortable: There is already a working version at commit [4817a49](https://github.com/scalableminds/webknossos/pull/7188/commits/4817a49853b8b5f122cf552a52b3b56e1b4f922d)

### Bugs:
- [x] rendering first segmentation layer and then color layer is broken :broken_heart: 

------
(Please delete unneeded items, merge only when none are left open)
- [x] Updated [changelog](../blob/master/CHANGELOG.unreleased.md#unreleased)

### CODE REVIEW

File: frontend/javascripts/oxalis/shaders/main_data_shaders.glsl.ts
Comment (philippotto) [inline frontend/javascripts/oxalis/shaders/main_data_shaders.glsl.ts:206]: do you think it's possible to generate the shader code so that it doesn't need to be recompiled when the layer order changes? on my machine, the shader compilation takes a few seconds which means that the UI is blocked everytime I change the layer order :S
`orderedColorLayerNames` can probably be passed as a uniform, but I'm not sure what needs to be done additionally (if it's possible at all).

File: frontend/javascripts/oxalis/view/left-border-tabs/layer_settings_tab.tsx
Comment (philippotto) [inline frontend/javascripts/oxalis/view/left-border-tabs/layer_settings_tab.tsx:384]:
```suggestion
    draggingDisabled: boolean = true,
```

would make the condition in 472 clearer, I think.

File: frontend/javascripts/oxalis/view/left-border-tabs/layer_settings_tab.tsx
Comment (philippotto) [inline frontend/javascripts/oxalis/view/left-border-tabs/layer_settings_tab.tsx:740]: why not a boolean initial value?

File: frontend/javascripts/oxalis/view/left-border-tabs/layer_settings_tab.tsx
Comment (philippotto) [inline frontend/javascripts/oxalis/view/left-border-tabs/layer_settings_tab.tsx:1117]:
```suggestion
          isColorLayer
```

File: frontend/javascripts/oxalis/geometries/materials/plane_material_factory.ts
Comment (philippotto) [inline frontend/javascripts/oxalis/geometries/materials/plane_material_factory.ts:636]: Is this ever written to?

File: frontend/javascripts/oxalis/model_initialization.ts
Comment (philippotto) [inline frontend/javascripts/oxalis/model_initialization.ts:811]: this guards against the case in which `colorLayerOrder` contains additional names that don't exist in the dataset, right? however, if this happens, couldn't it be that `onlyExistingLayers.length < colorLayerNames.length`? How about moving this filtering over the if-block in line 800?

**Review by philippotto — CHANGES_REQUESTED** [review]
nice stuff! DND works very nicely :+1:

I left some comments. The biggest one is with regard to the shader recompilation which I would like to avoid as much as possible.

And another thought: maybe the reordering UI should be disabled when the layer mode is additive because it won't make a difference in that case? Otherwise, I imagine this to be a bit misleading..

File: frontend/javascripts/oxalis/shaders/main_data_shaders.glsl.ts
Comment (philippotto) [reply frontend/javascripts/oxalis/shaders/main_data_shaders.glsl.ts:206]: I discussed this with Tom and we think that this shouldn't block the PR as the reordering won't be used with high frequency at the moment. Maybe we revamp the rendering architecture to be multi-pass soon. Then, this issue can be solved easier probably.

**Review by MichaelBuessemeyer — COMMENTED** [review]
@philippotto I think I addressed everything and the layer rendering is now only enabled when the cover blend mode Is turned on.

File: frontend/javascripts/oxalis/model_initialization.ts
Comment (philippotto) [inline frontend/javascripts/oxalis/model_initialization.ts:807]:
```suggestion
```

shouldn't be necessary, since onlyExistingLayers will be empty in that case.

**Review by philippotto — APPROVED** [review]
excellent :+1: just check my one suggestion before merging :)

**Pull request merged**
